### PR TITLE
[Builder] add global constant tensor frontend support and add test_constant_tensor to test_parser.py

### PR DESCRIPTION
**Pull request by Zhichenzzz**

### CODE REVIEW

File: allo/ir/builder.py [hunk 1]
```diff
@@ -424,6 +427,41 @@ def build_Assign(ctx, node):
         store_op = ASTTransformer.build_store(ctx, node.targets[0], rhs)
         return store_op
 
+    ## TODO: (zhichen) Int32 const tensor is done
```
Comment (chhzh123) [inline allo/ir/builder.py]: Delete this line

File: tests/test_parser.py [hunk 2]
```diff
@@ -736,6 +736,42 @@ def kernel() -> int32:
...
+
+
+def test_const_tensor_float():
+    def kernel() -> float32[2, 2]:
+        cp1: float32[2, 2] = [[1.1, 2.0], [3.0, 4.0]]
+        cp2: float32[2, 2] = [[1.1, 2.0], [3.0, 4.0]]
+        res: float32[2, 2] = 0.0
+        for i, j in allo.grid(2, 2):
+            res[i, j] = cp1[i, j] + cp2[i, j]
+        return res
```
Comment (chhzh123) [inline tests/test_parser.py]: Can you use a different kernel to better show your PR is general enough for different cases? For example, use a vector-multiply floating-point kernel with different sizes of matrices (larger than 2x2).

File: tests/test_parser.py [hunk 3]
```diff
@@ -736,6 +736,42 @@ def kernel() -> int32:
...
+
+    s = allo.customize(kernel)
+    f = s.build()
+    print(s.module)
+    np_0 = np.zeros((2, 2), dtype="float32")
+    np_1 = np.array([[1.1, 2.0], [3.0, 4.0]], dtype="float32")
+    np_0 = f()
+    assert np.array_equal(np_0, np_1 * 2.0)
```
Comment (chhzh123) [inline tests/test_parser.py]: Use `np.testing.assert_allclose` to test whether two **floating-point** tensors are equal. It is hard for floating-point numbers to be exactly the same due to numerical errors.

File: tests/test_parser.py [hunk 4]
```diff
@@ -764,3 +800,5 @@ def kernel() -> int32:
     test_compose_nested()
     test_no_init()
     test_double_partition()
+    test_const_tensor_int()
+    test_const_tensor_float
```
Comment (chhzh123) [inline tests/test_parser.py]: Missing parentheses.

File: allo/ir/builder.py [hunk 5]
```diff
@@ -537,7 +575,11 @@ def build_AnnAssign(ctx, node):
         ip = ctx.get_ip()
         type_hint = node.annotation
         if node.value is not None:
-            rhs = build_stmt(ctx, node.value)
+            if isinstance(node.value, (ast.List, ast.Tuple)):
+                rhs = ASTTransformer.build_constant_tensor(ctx, node)
+
+            elif isinstance(node.value, ast.Constant):
+                rhs = build_stmt(ctx, node.value)
```
Comment (chhzh123) [inline allo/ir/builder.py:580]: Add an `else` statement, probably raise an error?

File: allo/ir/builder.py [hunk 6]
```diff
@@ -557,13 +599,24 @@ def build_AnnAssign(ctx, node):
...
+                ctx.buffers[node.target.id] = rhs
+
+            elif isinstance(node.value, ast.Constant):
+                alloc_op = memref_d.AllocOp(memref_type, [], [], ip=ip)
+                alloc_op.attributes["name"] = StringAttr.get(node.target.id)
+                ctx.buffers[node.target.id] = alloc_op
+                if rhs is not None:
+                    with ip:
+                        # pylint: disable=unexpected-keyword-arg
+                        linalg_d.fill(rhs.result, outs=[alloc_op.result])
```
Comment (chhzh123) [inline allo/ir/builder.py:619]: Add `else`

File: allo/ir/builder.py (hunk 1 above)
Comment (Zhichenzzz) [reply allo/ir/builder.py]: Done

File: tests/test_parser.py (hunk 2 above)
Comment (Zhichenzzz) [reply tests/test_parser.py]: Oh, I think I can finish the global_vars call and generate random matrices, then test it by random matrix.

File: tests/test_parser.py (hunk 3 above)
Comment (Zhichenzzz) [reply tests/test_parser.py]: done

**Pull request merged**